src/Zend/Session.php

```diff
@@ -694,7 +694,22 @@ public static function writeClose($readonly = true)
             parent::$_writable = false;
         }
 
+        /*
+         * In some cases (when php processes run at the same time) a race condition can occur
+         * This error handler catches specific warnings or throws an exception otherwise
+         */
+        set_error_handler(function ($num, $str, $file, $line, $context = null) {
+            if (strpos($str, 'Failed to write session data using user defined save handler') !== false) {
+                parent::$_writable = false;
+            } else {
+                throw new Zend_Session_Exception($str);
```

Comment:
This essentially will change the behavior of this library a bit won't it?

Before it was throwing a PHP Warning (as the error handler is set to just act on Warnings), but now if the error message is not the specific "Failed to write session data using user defined save handler" message it'll throw an Exception. Most production systems just log a php warning, but don't halt execution. Throwing the exception here changes that, which I don't think this should be doing.

Possible to just act as before in the case that the warning message isn't the expected value?

Reply:
For what it's worth, maybe calling the previous error handler (returned by `set_error_handler`) would work here? I think it would have to be called after the new error handler is finished, so something from inside the error handler may have to be exposed to determine whether or not the previous one should be called.